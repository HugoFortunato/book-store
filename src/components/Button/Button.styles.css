.custom-button {
  padding: 10px 20px;
  color: white;
  background-color: #4caf50;

  font-size: 16px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

.custom-button:hover {
  background-color: #45a049;
}
