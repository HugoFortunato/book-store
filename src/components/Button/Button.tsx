import { ButtonTypes } from '.'
import './Button.styles.css'

export default function Button({ label, disabled, onClick }: ButtonTypes) {
  return (
    <button
      type="submit"
      className="custom-button"
      data-testid="button-id"
      disabled={disabled}
      onClick={onClick}
    >
      {label}
    </button>
  )
}
