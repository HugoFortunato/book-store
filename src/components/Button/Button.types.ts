import { ReactNode } from 'react'

export type ButtonTypes = {
  label: string
  children?: ReactNode
  disabled?: boolean
  onClick?: () => void
}
