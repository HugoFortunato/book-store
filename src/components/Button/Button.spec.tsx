import { render, fireEvent } from '@testing-library/react'

import '@testing-library/jest-dom'
import Button from './Button'

describe('<Button />', () => {
  const Test = () => {
    return (
      <Button
        color="#3498db"
        label="Click me"
        disabled
        data-testid="button-id"
      />
    )
  }

  it('should render a snapshot', () => {
    const { container } = render(<Test />)

    expect(container).toMatchSnapshot()
  })

  it('should test the clicked', () => {
    const { getByRole } = render(<Test />)
    const button = getByRole('button')

    fireEvent.click(button)

    expect(button).toHaveTextContent('Click me')
  })

  it('should have disabled property', () => {
    const { getByRole } = render(<Test />)

    const button = getByRole('button')

    expect(button).toHaveProperty('disabled', true)
  })

  it('should be disabled if specified', () => {
    const { getByRole } = render(
      <Button color="#3498db" label="Click me" disabled />,
    )
    const buttonElement = getByRole('button')

    expect(buttonElement).toBeDisabled()
  })

  it('should call the onClick callback when clicked', () => {
    const mockClickHandler = jest.fn()

    const { getByRole } = render(
      <Button color="#3498db" label="Click me" onClick={mockClickHandler} />,
    )
    const buttonElement = getByRole('button')

    fireEvent.click(buttonElement)

    expect(mockClickHandler).toHaveBeenCalledTimes(1)
  })
})
