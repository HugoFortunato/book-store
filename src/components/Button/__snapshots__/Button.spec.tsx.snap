// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a snapshot 1`] = `
<div>
  <button
    class="custom-button"
    data-testid="button-id"
    disabled=""
    type="submit"
  >
    Click me
  </button>
</div>
`;
